sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java

```diff
@@ -0,0 +1,269 @@
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<UploadManifestResult> uploadManifestWithResponse(BinaryData data, Context context) {
+        return this.asyncClient.uploadManifestWithResponse(data.toFluxByteBuffer(), context).block();
+    }
+
+    /**
+     * Uploads a blob to the repository.
+     * The client currently uploads the entire blob\layer as a single unit.
+     * <p>
+     * Also, the blob is read into memory and then an upload operation is performed as a single operation.
+     * We currently do not support breaking the layer into multiple chunks and uploading them one at a time
+     *
+     * @param data The blob\image content that needs to be uploaded.
+     * @return The operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public UploadBlobResult uploadBlob(BinaryData data) {
+        return this.asyncClient.uploadBlob(data).block();
+    }
+
+    /**
+     * Uploads a blob to the repository.
+     * The client currently uploads the entire blob\layer as a single unit.
+     * <p>
+     * Also, the blob is read into memory and then an upload operation is performed as a single operation.
+     * We currently do not support breaking the layer into multiple chunks and uploading them one at a time
+     * The service does support this via range header.
+     *
+     * @param data The blob\image content that needs to be uploaded.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return The rest response containing the operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<UploadBlobResult> uploadBlobWithResponse(BinaryData data, Context context) {
+        return this.asyncClient.uploadBlobWithResponse(data.toFluxByteBuffer(), context).block();
+    }
+
+    /**
+     * Download the manifest associated with the given tag or digest.
+     * We currently only support downloading OCI manifests.
+     *
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param tagOrDigest The tag or digest of the manifest.
+     * @return The manifest associated with the given tag or digest.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code tagOrDigest} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public OciManifest downloadManifest(String tagOrDigest) {
```

Comment:
To be consistent with async client, we should use `digest` here and in other places where `tagOrDigest` is used.
```suggestion
    public OciManifest downloadManifest(String digest) {
```

Reply:
I will actually change other places to tagOrDigest since that is what is most consistent across all clients.